kidkode/src/main/webapp/script.js

```diff
@@ -139,40 +144,81 @@ startQuiz();
 ////////////////
 // PAGINATION //
 ////////////////
-/*
- * The following functions call the showSlide() function with the next or
- * previous slide.
- */
-function showNextSlide() {
-  showSlide(currentSlide + 1);
-}
-
-function showPreviousSlide() {
-  showSlide(currentSlide - 1);
-}
-
 /*
  * This function shows and hides the appropriate slides and buttons depending
  * on which slide is active and which aren't.
  */
-function showSlide(n) {
-  slides[currentSlide].classList.remove("active-slide");
-  slides[n].classList.add("active-slide");
-  currentSlide = n;
-  if (currentSlide === 0) {
-    previousButton.style.display = "none";
+function showSlide(newStep, move) {
+  var toRemove = document.getElementById(currentStep);
+  if (newStep != null) {
+    // remove current slide as active slide
+    toRemove.classList.remove("active-slide");
+    // assign new slide as active slide
+    var newSlide = document.getElementById(newStep);
+    newSlide.classList.add("active-slide");
+    // update the current step
+    currentStep = newStep;
   } else {
-    previousButton.style.display = "inline-block";
+    // if the user reached the end of the quiz
+    toRemove.classList.remove("active-slide");
   }
-  if (currentSlide === slides.length - 1) {
+  if ((move === null || move === "previous") && newStep === "step0") {
+    // if new slide is the first slide
+    previousButton.style.display = "none";
+    nextButton.style.display = "inline-block";
+    submitButton.style.display = "none";
+  } else if (move === "next" && newStep === null) {
+    // if new slide is the last slide
+    previousButton.style.display = "inline-block";
     nextButton.style.display = "none";
     submitButton.style.display = "inline-block";
   } else {
+    // if new slide is any slide in between
+    previousButton.style.display = "inline-block";
     nextButton.style.display = "inline-block";
     submitButton.style.display = "none";
   }
 }
 
+/*
+ * This function returns the next property of the current question, based
+ * on the choice the user picked in this question. As of now, the choice is hardcoded.
+ */
+function getNext(testQuestions, currentStep, userChoice) {
+  for (let [step, stepItems] of Object.entries(testQuestions)) {
+    if (step === currentStep) {
+      for (var option in stepItems.answers) {
+        if (option === userChoice) {
+          return stepItems.answers[option].next;
+        }
+      }
+    }
+  }
+}
+
+/*
+ * This function shows the next slide, based on the current question and the choice
+ * the user picked. As of now, the choice is hardcoded.
+ */
+function showNextSlide() {
+  fetch(jsonFile)
```

Comment:
There's probably a way we can avoid fetching the JSON multiple times but I think we should worry about that at the end if we have time

Reply:
Yeah, I agree... because fetching the JSON does cause a teeny bit of a delay when loading the next question. Hopefully we'll have some time to look over how we can make this more efficient!